library/core/src/intrinsics.rs

```diff
@@ -1918,6 +1918,12 @@ extern "rust-intrinsic" {
     #[rustc_const_unstable(feature = "const_heap", issue = "79597")]
     pub fn const_allocate(size: usize, align: usize) -> *mut u8;
 
+    /// Deallocate a memory which allocated by `intrinsics::const_allocate` at compile time.
+    /// Should not be called at runtime.
```

Comment:
IMO it would make more sense to just make this a NOP at runtime (since it looks like it will also be a NOP for some compile-time cases, so this is more consistent).

Reply:
We already have const_eval_select for that. The intrinsic should imo not be implemented in backends

Reply:
> We already have const_eval_select for that.

I had thought wrapping `const_[de]allocate` with  `const_eval_select` works well, but even just defining an function which calls the intrinsic causes a codegen error. So we may need to add nop to codegens anyway.

Reply:
That seems strange, the const function shouldn't even be codegen'd... right?

Reply:
It's in the generic args of the `const_eval_select` call

Reply:
https://play.rust-lang.org/?version=nightly&mode=debug&edition=2021&gist=7c6e75dd1cc1d2f14878ee24a3fe43cb
It occurs when compiling as a library crate.